src/OptimalControlUtilities/include/BipedalLocomotionControllers/OptimalControlUtilities/PID.h

```diff
@@ -0,0 +1,205 @@
+
+    iDynTree::Rotation m_desiredOrientation; /**< Desired orientation. */
+    iDynTree::Rotation m_orientation; /**< Actual orientation. */
+
+    /**
+     * Transform a 3x3 matrix into a skew-symmetric matrix.
+     * @param input is a 3x3 matrix;
+     * @return a 3x3 skew-symmetric matrix
+     */
+    iDynTree::Matrix3x3 skewSymmetric(const iDynTree::Matrix3x3& input);
+
+    /**
+     * Evaluate control
+     */
+    void evaluateControl() final;
+
+public:
+    /**
+     * Set rotational PID Gains
+     * @param c0 pid gain;
+     * @param c1 pid gain;
+     * @param c2 pid gain.
+     */
+    void setGains(const double& c0, const double& c1, const double& c2);
+
+    /**
+     * Set the desired trajectory.
+     * @param desiredAcceleration desired acceleration (rad/s^2);
+     * @param desiredVelocity desired velocity (rad/s);
+     * @param desiredOrientation rotation matrix
+     */
+    void setDesiredTrajectory(const iDynTree::Vector3& desiredAcceleration,
+                              const iDynTree::Vector3& desiredVelocity,
+                              const iDynTree::Rotation& desiredOrientation);
+
+    /**
+     * Set feedback
+     * @param velocity angular velocity:
+     * @param orientation rotation matrix.
+     */
+    void setFeedback(const iDynTree::Vector3& velocity, const iDynTree::Rotation& orientation);
+};
+
+/**
+ * Standard Liner position PID
+ */
+class PositionPID : public CartesianPID
+{
+    iDynTree::Vector3 m_kp; /**< Proportional gain */
+    iDynTree::Vector3 m_kd; /**< Derivative gain */
+
+    iDynTree::Vector3 m_desiredPosition; /**< Desired position. */
+
+    iDynTree::Vector3 m_position; /**< Actual position. */
+
+    iDynTree::Vector3 m_error; /**< Position error */
+    iDynTree::Vector3 m_dotError; /**< Velocity error */
+
+    /**
+     * Evaluate control
+     */
+    void evaluateControl() final;
+
+public:
+    /**
+     * Set PID Gains
+     * @param kp proportional gain (scalar);
+     * @param kd derivative gain (scalar).
+     */
+    void setGains(const double& kp, const double& kd);
+
+    /**
+     * Set PID Gains
+     * @param kp proportional gain (vector);
+     * @param kd derivative gain (vector).
+     */
+    void setGains(const iDynTree::Vector3& kp, const iDynTree::Vector3& kd);
+
+    /**
+     * Set the desired trajectory.
+     * @param desiredAcceleration desired acceleration (m/s^2);
+     * @param desiredVelocity desired velocity (m/s);
+     * @param desiredPosition desired position (m).
+     */
+    void setDesiredTrajectory(const iDynTree::Vector3& desiredAcceleration,
+                              const iDynTree::Vector3& desiredVelocity,
+                              const iDynTree::Vector3& desiredPosition);
+
+    /**
+     * Set feedback
+     * @param velocity linear velocity:
+     * @param position actual position.
+     */
+    void setFeedback(const iDynTree::Vector3& velocity, const iDynTree::Vector3& position);
+};
+
+class OneDegreePID
```

Comment:
``OneDegree`` celsius :snowman_with_snow: ? Maybe ``SingleDoFPID`` or similar may be more descriptive.

Reply:
`OneDegreePID` has been removed in https://github.com/dic-iit/bipedal-locomotion-controllers/commit/8a90fd2c8fb7160a20be9913fb24de80262d96c1